packages/devtools/lib/src/timeline/timeline_protocol.dart

```diff
@@ -573,6 +812,22 @@ class TimelineEvent {
     }
   }
 
+  void formatFromRoot(StringBuffer buf, String indent) {
+    getRoot().format(buf, indent);
+  }
+
+  void writeTraceToBuffer(StringBuffer buf) {
+    if (eventTraces.isNotEmpty) {
+      buf.writeln(eventTraces.first.toString());
```

Comment:
why is first special cased? this output format seems hard for someone who is not familiar with it to understand.